app/controllers/applicant/LeadPartnerEntityController.scala

```diff
@@ -59,7 +59,7 @@ class LeadPartnerEntityController @Inject()(val authConnector: AuthConnector,
           formWithErrors => Future.successful(BadRequest(leadPartnerEntityPage(formWithErrors))),
           partyType =>
             for {
-              _ <- sessionService.cache[PartyType](leadPartnerEntityKey, partyType)
+              _ <- partnersService.upsertPartner(profile.registrationId, 1, Entity(None, partyType = partyType, Some(true), None))
```

Comment:
Ideally we don't want to completely reset a model like that (if it's a returning user). If the answer is unchanged it shouldn't set other fields to none.